engine/utils/engine_matcher_utils.h

```diff
@@ -0,0 +1,195 @@
+#include <algorithm>
+#include <iostream>
+#include <iterator>
+#include <regex>
+#include <string>
+#include <vector>
+#include "utils/cpuid/cpu_info.h"
+
+namespace engine_matcher_utils {
+// for testing purpose
+const std::vector<std::string> cortex_llamacpp_variants{
+    "cortex.llamacpp-0.1.25-25.08.24-linux-amd64-avx-cuda-11-7.tar.gz",
+    "cortex.llamacpp-0.1.25-25.08.24-linux-amd64-avx-cuda-12-0.tar.gz",
+    "cortex.llamacpp-0.1.25-25.08.24-linux-amd64-avx.tar.gz",
+    "cortex.llamacpp-0.1.25-25.08.24-linux-amd64-avx2-cuda-11-7.tar.gz",
+    "cortex.llamacpp-0.1.25-25.08.24-linux-amd64-avx2-cuda-12-0.tar.gz",
+    "cortex.llamacpp-0.1.25-25.08.24-linux-amd64-avx2.tar.gz",
+    "cortex.llamacpp-0.1.25-25.08.24-linux-amd64-avx512-cuda-11-7.tar.gz",
+    "cortex.llamacpp-0.1.25-25.08.24-linux-amd64-avx512-cuda-12-0.tar.gz",
+    "cortex.llamacpp-0.1.25-25.08.24-linux-amd64-avx512.tar.gz",
+    "cortex.llamacpp-0.1.25-25.08.24-linux-amd64-noavx-cuda-11-7.tar.gz",
+    "cortex.llamacpp-0.1.25-25.08.24-linux-amd64-noavx-cuda-12-0.tar.gz",
+    "cortex.llamacpp-0.1.25-25.08.24-linux-amd64-noavx.tar.gz",
+    "cortex.llamacpp-0.1.25-25.08.24-linux-amd64-vulkan.tar.gz",
+    "cortex.llamacpp-0.1.25-25.08.24-mac-amd64.tar.gz",
+    "cortex.llamacpp-0.1.25-25.08.24-mac-arm64.tar.gz",
+    "cortex.llamacpp-0.1.25-25.08.24-windows-amd64-avx-cuda-11-7.tar.gz",
+    "cortex.llamacpp-0.1.25-25.08.24-windows-amd64-avx-cuda-12-0.tar.gz",
+    "cortex.llamacpp-0.1.25-25.08.24-windows-amd64-avx.tar.gz",
+    "cortex.llamacpp-0.1.25-25.08.24-windows-amd64-avx2-cuda-11-7.tar.gz",
+    "cortex.llamacpp-0.1.25-25.08.24-windows-amd64-avx2-cuda-12-0.tar.gz",
+    "cortex.llamacpp-0.1.25-25.08.24-windows-amd64-avx2.tar.gz",
+    "cortex.llamacpp-0.1.25-25.08.24-windows-amd64-avx512-cuda-11-7.tar.gz",
+    "cortex.llamacpp-0.1.25-25.08.24-windows-amd64-avx512-cuda-12-0.tar.gz",
+    "cortex.llamacpp-0.1.25-25.08.24-windows-amd64-avx512.tar.gz",
+    "cortex.llamacpp-0.1.25-25.08.24-windows-amd64-noavx-cuda-11-7.tar.gz",
+    "cortex.llamacpp-0.1.25-25.08.24-windows-amd64-noavx-cuda-12-0.tar.gz",
+    "cortex.llamacpp-0.1.25-25.08.24-windows-amd64-noavx.tar.gz",
+    "cortex.llamacpp-0.1.25-25.08.24-windows-amd64-vulkan.tar.gz",
+};
+const std::vector<std::string> cortex_onnx_variants{
+    "cortex.onnx-0.1.7-windows-amd64.tar.gz"};
+
+const std::vector<std::string> cortex_tensorrt_variants{
+    "cortex.tensorrt-llm-0.0.9-linux-cuda-12-4.tar.gz",
+    "cortex.tensorrt-llm-0.0.9-windows-cuda-12-4.tar.gz"};
+
+inline std::string GetSuitableAvxVariant();
```

Comment:
Can remove this because we've already had the implementation at the same header file